### ISSUE TITLE
aiofiles: 'async with' received an object from __aenter__ that does not implement __await__: generator

### ISSUE THREAD

**Issue by striggldy** [Nuitka/Nuitka#400]
```
$ python -m nuitka --version
0.6.4rc5
Python: 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 22:22:05) [MSC v.1916 64 bit (AMD64)]
Executable: F:\frameworks\python371x64\python.exe
OS: Windows
Arch: x86_64
```

aiofiles: https://github.com/Tinche/aiofiles
```
$ pip list | grep aiofiles
aiofiles         0.5.0.dev0
```

Reproducer:
```
$ cat aiofiles_test.py
import asyncio
import aiofiles

async def main():
    async with aiofiles.open("foo.txt", "w") as fh:
        await fh.write("foo")

if __name__ == '__main__':
    loop = asyncio.get_event_loop()
    loop.run_until_complete(main())
```

built with
```
$ python -m nuitka --mingw --remove-output --nofollow-imports aiofiles_test.py
```

produces
```
$ ./aiofiles_test.exe
...
    async with aiofiles.open("foo.txt", "w") as fh:
TypeError: 'async with' received an object from __aenter__ that does not implement __await__: generator
```

**Comment by kayhayen** [comment]
I managed to reproduce this one, same error on Linux.

**Comment by kayhayen** [comment]
Seems the new 3.7 feature to give better error messages for async with errors was misunderstood by me, there was code checking the async awaitable result and insist it must be async iterable, but generators are not.

Nothing in the test suite caught this, which was disabled for 3.7 still for no reason, but passed just as well as it does with 3.6, so there is no coverage of that behavior. Anyway, should be good now. It's on the factory branch.

**Comment by kayhayen** [comment]
So this is in 0.6.5rc3, currently develop branch.

**Comment by levesquejf** [comment]
@kayhayen We hit the same bug. When do you expect 0.6.5 to be available?

**Comment by kayhayen** [comment]
@levesquejf Ironically I am waiting for me to find the time to hunt down the other bug you found, which I find embarassing. But if you say it's better to have this now, I will commence the release process early next week.

Yours,
Kay

**Comment by kayhayen** [comment]
Released as 0.6.5
